content/swiftui/concepts/notifications/notifications.md

```diff
@@ -0,0 +1,190 @@
+
+*This code will provide the user with options to select their preferred sorting order.*
+
+
+## 3. Toasts (Message Overlays)
+A toast is a small, temporary notification that appears at the bottom of the screen. It's typically used to display non-intrusive messages or status updates.
+
+// Message Overlay. Example:
+
+```struct ContentView: View {
+    @State private var showToast = false
+
+    var body: some View {
+        Button("Show Toast") {
+            showToast = true
+        }
+        .toast(isPresented: $showToast, content: {
+            ToastView(message: "Data saved successfully")
+        })
+    }
+}
+
+struct ToastView: View {
+    let message: String
+
+    var body: some View {
+        VStack {
+            Text(message)
+                .padding()
+                .background(Color.black.opacity(0.7))
+                .foregroundColor(.white)
+                .cornerRadius(10)
+        }
+        .frame(maxWidth: .infinity, alignment: .center)
+        .padding()
+        .animation(.easeInOut(duration: 0.3))
+    }
+}
+```
+
+>> `@State private var showToast = false`: This creates a state variable named showToast, which determines whether the toast should be shown or not.
+
+>> `Button("Show Toast") { showToast = true }`: A button, when tapped, sets the showToast state variable to true, triggering the display of the toast.
+
+>> `.toast(isPresented: $showToast, content: { ... })`: A custom view modifier (implemented as an extension) that displays a toast when the showToast state is true. The content closure defines the content of the toast, which is an instance of the `ToastView`` view.
+
+>> `ToastView`: A custom view that represents the toast. It contains a Text view displaying the provided message, with a dark background and white text color. The .frame, .padding, and .animation modifiers are used to control the appearance and animation of the toast.
+
+*This code will show a short confirmation message after the user successfully saves some data.*
+
+## 4. Custom Notifications 
+
+We can create our own custom notifications by designing custom views and animations. These can be used to provide unique and engaging user interface elements tailored to your app's style and branding.
+
+Custom notifications allow us to get creative with how we present information or engage users within our SwiftUI app.
+
+The specific implementation and appearance of notifications can be customized based on our app's design and user experience goals. SwiftUI's flexible and declarative approach makes it easy to create engaging and user-friendly notifications that enhance your app's overall experience.
+
+```struct ContentView: View {
+    var body: some View {
+        VStack {
+            Text("Welcome to My App")
+                .font(.title)
+            
+            Button("Show Custom Notification") {
+                // Trigger custom notification display
+            }
+            .padding()
+        }
+    }
+}
+```
```

Comment:
````suggestion
```swift
struct ContentView: View {
    var body: some View {
        VStack {
            Text("Welcome to My App")
                .font(.title)
            
            Button("Show Custom Notification") {
                // Trigger custom notification display
            }
            .padding()
        }
    }
}
```
````